mrmustard/lab_dev/circuit_components.py

```diff
@@ -182,6 +164,32 @@ def __getitem__(self, idx: Union[Mode, Sequence[Mode]]):
         ret._wires = self._wires[idx]
         ret._parameter_set = self.parameter_set
         return ret
+    
+    def __matmul__(self, other: CircuitComponent) -> CircuitComponent:
```

Comment:
This errors:

`from mrmustard.lab_dev.transformations import *`
`d0 = Dgate(1, modes=[0])`
`d0 @ d0.adjoint`